core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/browser/userAgentParser.ts

```diff
@@ -0,0 +1,242 @@
+export interface ParsedUserAgent {
+    browser: {
+        name: string;
+        version: string;
+    };
+    device: {
+        type: 'mobile' | 'tablet' | 'desktop';
+        vendor?: string;
+        model?: string;
+    };
+    os: {
+        name: string;
+        version?: string;
+    };
```

Comment:
Let's create a type for every property, to prevent issues in the future and better mocking